src/python/CMSMonitoring/StompAMQ.py

```diff
@@ -44,17 +47,19 @@ def validate(doc, schema='auto', verbose=False):
                 logging.warning('Local schema {} is not json compliant'.format(schema))
 
     if schema in _local_schemas:
-        return validate_schema(_local_schemas[schema], doc, verbose)
+        return validate_schema(_local_schemas[schema], doc, loglevel)
 
     elif schema in _schemas.schemas():
-        return validate_schema(_schemas.schemas()[schema], doc, verbose)
+        return validate_schema(_schemas.schemas()[schema], doc, loglevel)
 
     else:
-        for sname, sch in _schemas.schemas().items():
-            if validate_schema(sch, doc, verbose):
-                return True
+        logging.warning('No validation schema provided, comparing all docs to the first one')
+        with open('auto_schema.json', 'w') as schfile:
+            json.dump(doc, schfile, indent=2, sort_keys=True)
+        _local_schemas['auto'] = doc
+        return [], []
```

Comment:
Another issue, you're mixing return data-types, here it is list, list, while other returns None, None. It is better to have consistent return data-type.

Reply:
You have a better suggestion? The point is to distinguish between the case where the validation was successful and the one where none was run. Note that the `None, None` only appears if people don't run validation. In that case they'll hopefully not use the validation result in the first place.